web/src/engine/websites/XCaliBRScans.ts

```diff
@@ -108,31 +105,42 @@ export default class extends DecoratableMangaScraper {
     }
 
     public override async FetchImage(page: Page, priority: Priority, signal: AbortSignal): Promise<Blob> {
-        const blobMainImage = await Common.FetchImageAjax.call(this, page, priority, signal);
-
         switch (page.Parameters.scrambled) {
-            case 0: // No scrambling, return image
-                return blobMainImage;
-            case 1: // Flip picture
+            case 0: {// No scrambling, return image {
+                return await Common.FetchImageAjax.call(this, page, priority, signal);
+            }
+            case 1: {// Flip picture
+                const blobMainImage = await Common.FetchImageAjax.call(this, page, priority, signal);
                 return DeScramble(blobMainImage, async (bitmap, ctx) => {
                     ctx.scale(-1, 1);
                     ctx.drawImage(bitmap, 0, 0, -bitmap.width, bitmap.height);
                 });
-            case 2: // Combine/Flip 2 pictures
-            {
-                const pageUrl = (page.Parameters.secondaryPic) as string;
-                const request = new Request(pageUrl, { headers: { Referer: this.URI.href } });
-                const response = await Fetch(request);
-                const b1 = await createImageBitmap(blobMainImage);
-                const b2 = await createImageBitmap(await response.blob());
-                return DeScramble(new ImageData(b1.width + b2.width, b1.height), async (_, ctx) => {
-                    ctx.scale(-1, 1);
-                    ctx.drawImage(b2, 0, 0, -b2.width, b2.height);
-                    ctx.drawImage(b1, -b2.width, 0, -b1.width, b1.height);
-                });
             }
-            default:
-                throw Error();
+            case 2: { // Combine + Flip 2 pictures
+                return this.imageTaskPool.Add(async () => {
+                    const blobMainImage = await this.FetchBlob(page.Link.href, signal);
```

Comment:
Same as above, two downloads in a single task will affect the meaning of rate limit.